openhcl/openhcl_boot/src/arch/x86_64/tdx.rs

```diff
@@ -98,6 +172,26 @@ impl minimal_rt::arch::IoAccess for TdxIoAccess {
     }
 }
 
+/// Invokes a hypercall via a TDCALL
+///
+/// # Safety
+///
+/// The caller ensures that the I/O pages are valid and not concurrently accessed
+pub unsafe fn invoke_tdcall_hypercall(
+    control: hvdef::hypercall::Control,
+    input_page: u64,
+    output_page: u64,
+) -> hvdef::hypercall::HypercallOutput {
+    let result = tdcall_hypercall(&mut TdcallInstruction, control, input_page, output_page);
```

Comment:
hm, i wonder if td_hypercall also needs to be an unsafe function now with input_page and output_page

Reply:
Maybe, we should add a hypercall option that takes in/out pages that is unsafe? and then the fast hypercall path is safe since there's no outside refs to memory.

Reply:
Or could we instead return a newtype from `initialize_hypercalls` that helps enforce the safety constraints for in/out pages, that only `initialize_hypercalls` returns? 

I still think that the `tdcall_hypercall` that accepts in/out pages must be unsafe, though. But having the newtype would allow you to not require this to be an unsafe function.